staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -390,9 +429,14 @@ func (crEncoder) Encode(obj runtime.Object, w io.Writer) error {
 	}
 }
 
-type UnstructuredCreator struct{}
+type unstructuredCreator struct {
+	delegate runtime.ObjectCreater
+}
 
-func (UnstructuredCreator) New(kind schema.GroupVersionKind) (runtime.Object, error) {
+func (c unstructuredCreator) New(kind schema.GroupVersionKind) (runtime.Object, error) {
+	if obj, err := c.delegate.New(kind); err == nil {
```

Comment:
Do you mean a CRD with Kind `ListOptions` will break? I hadn't thought about that to be honest.

Would it help if I remove the delegation for `New()` and only keep it for the methods where we can distinguish whether the caller specifically expects `Unstructured`? The particular `DeleteOptions` code path we care about here probably doesn't call `New()` anymore now that I've fixed the decoder to not return `Unstructured`.